### PR TITLE
Add version int tuple `__version_info__` to JAX

### PR DESCRIPTION
**Pull request by sharadmv**

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Looks good to me!

File: jax/version.py [hunk 1]
```diff
@@ -13,5 +13,7 @@
 # limitations under the License.
 
 __version__ = "0.3.5"
+__version_info__ = tuple(map(int, __version__.split(".")))
```
Comment (yashk2810) [inline jax/version.py:16]: Maybe use `.isdigit()`?

Comment (yashk2810) [reply jax/version.py:16]: `tuple(int(i) for i in __version__.split(".") if i.isdigit())`

Comment (sharadmv) [reply jax/version.py]: Done.

File: jax/version.py [hunk 2]
```diff
@@ -13,5 +13,7 @@
 # limitations under the License.
 
 __version__ = "0.3.5"
+__version_info__ = tuple(int(i) for i in __version__.split(".") if i.isdigit())
 
 _minimum_jaxlib_version = "0.3.0"
+_minimum_jaxlib_version_info = tuple(int(i) for i in _minimum_jaxlib_version.split(".") if i.isdigit())
```
Comment (yashk2810) [inline jax/version.py:19]: Instead of repeating this twice, maybe make a private function?

Comment (sharadmv) [reply jax/version.py]: Done.

**Pull request merged**